node/type.go

```diff
@@ -0,0 +1,32 @@
+package node
+
+// node.Type defines a type for Nodes to be differentiated.
+type Type uint8
```

Comment:
can we call this `Mode` or `RunMode` or something like that? I don't really mind, I just think `Type` is too vague. Also once we pick this term, we should stick to using it throughout the entire codebase.

Reply:
Let's move this discussion to https://github.com/celestiaorg/celestia-node/issues/41. but in general, I am inclined to agree with Mode instead of Type